compiler/front/options.nim

```diff
@@ -508,12 +508,12 @@ proc report*(conf: ConfigRef, id: ReportId): TErrorHandling =
   ## .. note:: This check is done only for reports that are generated via
   ##           IDs, because that's the only way report can (supposedly)
   ##           enter the error message system twice.
-  return conf.report(conf.m.reports.getReport(id))
+  result = conf.report(conf.m.reports.getReport(id))
 
 proc report*(conf: ConfigRef, node: PNode): TErrorHandling =
   ## Write out report from the nkError node
   assert node.kind == nkError
-  return conf.report(node.reportId)
+  result = conf.report(node.reportId)
```

Comment:
Instead, report freeing should be placed here, as we have the ID information and can access the report store directly. Requiring the *formatting* function to *cleanup* the report is not ideal.